src/checker/mod.rs

```diff
@@ -102,17 +104,34 @@ impl Checker for Checkers {
     where
         'a: 's,
     {
-        let mut collective = Vec::<Suggestion<'s>>::with_capacity(chunks.len());
+        let mut collective = HashSet::<Suggestion<'s>>::new();
         if let Some(ref hunspell) = self.hunspell {
             collective.extend(hunspell.check(origin, chunks)?);
         }
         if let Some(ref nlprule) = self.nlprules {
             collective.extend(nlprule.check(origin, chunks)?);
         }
 
-        collective.sort();
+        let mut suggestions: Vec<Suggestion<'s>> = Vec::from_iter(collective);
+        suggestions.sort();
+        if suggestions.len() == 0 {
+            return Ok(suggestions);
+        }
+
+        // Iterate through suggestions and identify overlapping ones.
+        let _omitted = suggestions
+            .iter()
+            .enumerate()
+            .filter_map(|(idx, cur)| {
+                if idx > 0 && cur.is_overlapped(&suggestions[idx - 1]) {
+                    Some(cur.chunk.clone())
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>();
```

Comment:
This doesn't work like you expect :)

`.iter()` takes a immutable reference and hence cannot modify `suggestions`.

```suggestion
        let suggestions = Vec::from_iter(suggestions
            .into_iter()
            .enumerate()
            .filter_map(|(idx, cur)| {
                if idx == 0 || !cur.is_overlapped(&suggestions[idx - 1]) {
                    Some(cur.chunk)
                } else {
                    None
                }
            }));
```